More content in the home page (#168)
awslabs/amazon-s3-find-and-forget: README.md

    <img src="https://codecov.io/gh/awslabs/amazon-s3-find-and-forget/branch/master/graph/badge.svg">
</h1>

Amazon S3 Find and Forget is a solution to the need to selectively erase records
from data lakes stored on Amazon Simple Storage Service (Amazon S3). This
solution can assist data lake operators to handle data erasure requests, for
example, pursuant to the European General Data Protection Regulation (GDPR).

The solution can be used with Parquet-format data stored in Amazon S3 buckets.
Your data lake is connected to the solution via AWS Glue tables and by
specifying which columns in the tables contain user identifiers.

Once configured, you can queue record identifiers that you want the
corresponding data erased for. You can then run a deletion job to remove the
data corresponding to the records specified from the objects in the data lake. A
report log is provided of all the S3 objects modified.

## Installation

The solution is available as an AWS CloudFormation template and should take
about 20 to 40 minutes to deploy. See the

> [deployment guide](docs/USER_GUIDE.md#deploying-the-solution) to launch the

Author: Done

deployment with just one click and the
[cost overview guide](docs/COST_OVERVIEW.md) to learn about its costs.

> Warning: This project is currently in beta release
>
> - Actively developed by AWS
> - Carefully review code and documentation if you intend to use the solution
>   for production purposes
> - Get in touch with the team via
>   [Github Issues](https://github.com/awslabs/amazon-s3-find-and-forget/issues)

> >   for providing feedback and to ask any question. We appreciate your help

Author: I tried to provide more comprehensive details about what beta means for us.

## Usage

The solution provides a web user interface, and a REST API to allow you to
integrate it in your own applications.

> ![HomePage Screenshot](docs/images/home-screenshot.png)

Author: Not sure about this one. The blogpost may have more details about this, so for now I just put one screenshot. But I can also remove it if we think it doesn't add any value

Reviewer: That screenshot on its own doesn't feel very valuable to me.

I have no problem with adding it now, but I think we should improve it ... maybe experiment with creating an animated GIF that shows the typical workflow?

Author: I think you're right. I removed it for now, let's try making a gif in a separate PR

See the [user guide](docs/USER_GUIDE.md) to learn how to use the solution and
the [API Specification](docs/api/README.md) to integrate the solution to your
own applications.

## Architecture

The goal of the solution is to provide a secure, reliable, performant and cost
effective tool for finding and removing individual records within objects stored
in S3 buckets. In order to achieve these goals the solution has adopted the

> following design principles:

Author: The design principles in the architecture page are actually a bit of a mixture of design principles and consequences of some of our design principles. I tried to re-organise them here. If this makes sense, then I can also copy and paste there.

1. **Security first:** Every component implements least privilege access,
   encryption is performed at all layers at rest and in transit, authentication
   is provided out of the box, expiration of logs is easily configurable, and

>    private data is automatically obfuscated or irreversibly deleted as soon as

Author: what about MatchIDs

>    private data is automatically obfuscated or irreversibly deleted as soon as

Reviewer 2: yeah Match IDs is probably more consistent - have we introduced that term by this point in the document though?

>    private data is automatically obfuscated or irreversibly deleted as soon as

Author: We didn't, so I introduced the term there, hopefully it makes sense now

   possible when persisting state.
2. **Built to scale:** The system has been designed and battle tested for
   performance and high availability with Data Lakes of TB/PB size.
3. **Cost optimised:**
   - **Perform work in batches:** Since the time complexity of removing a single

>      vs multiple records in a single object is practically equal and it is

Reviewer 2: ```suggestion
     vs multiple matches in a single S3 object is practically equal, and it is
```

Author: Not sure, do you think that match could be confused with the idea of actually just removing that value and leaving the rest of the row there? Perhaps we can use `rows`?

>      common for data owners to have the legal requirement of removing data

Reviewer 2: ```suggestion
     common for data owners to have the requirement of removing data
```

>      within a given _timeframe_, the solution is designed to allow the customer

Reviewer 2: ```suggestion
     within a given _timeframe_, the solution is designed to allow the solution operator
```

     to "queue" multiple matches to be removed in a single job.
   - **Fail fast:** A deletion job takes place in two distinct phases: Find and
     Forget. The Find phase queries the objects in your S3 data lakes to find

>      any objects which contain records where a specified column contains at

Reviewer 2: ```suggestion
     any objects which contain matches where a specified column contains at
```

Reviewer 2: or records might be fine, I think we're just jumping between terms a bit

Author: yeah, I think records is what we were using to call "rows" as opposite of match which is only the value for that column?

Reviewer 2: yeah that sounds right

     least one of record identifiers (known as **Match IDs**) in the deletion
     queue. If any queries fail, the job will abandon as soon as possible and
     the Forget phase will not take place. The Forget Phase takes the list of

>      objects returned from the Find phase, and deletes the relevant rows in only

Reviewer 2: ```suggestion
     objects returned from the Find phase, and deletes only the relevant rows in
```

     those objects.

>    - **Find, then Forget:** The process of separating the two phases, as

Reviewer 2: I don't think Find, then Forget is the right subheader for this section. I'm wondering if we're just restating the benefit of data formats like Parquet here, which won't necessarily hold true for other data formats like CSV/JSON?

Author: Ok, perhaps we can rephrase the subheader as `Optimised for Parquet`?

Reviewer 2: yeah either that or "Optimised for Columnar Data Formats"

     opposite of processing the read/write on an object basis, is another cost

>      optimisation technique based on the fact columnar dense formats such as

Reviewer 2: ```suggestion
     optimisation technique leveraging columnar dense formats such as
```

>      Parquet allow to find matches with increased performance and cost

Reviewer 2: ```suggestion
     Parquet allow to efficiently find matches
```

>      optimisation by reading only relevant columns instead of full content from

Reviewer 2: ```suggestion
    by reading only relevant columns instead of full content from
```

     all columns. By processing only the relevant columns during the Find and

>      deletegating it to a serverless managed service such as Amazon Athena, we

Reviewer 2: ```suggestion
     delegating the queries to a serverless managed service such as Amazon Athena, we
```

     can then process full rows only during the Forget phase, restricting the
     most expensive operation to only the relevant objects.

>    - **Serverless:** Where possible, the solution only uses Serverless

Reviewer 2: we should probably state why Serverless optmizes costs here

Author: Done

     components. All the components for Web UI, API and Deletion Jobs are
     Serverless (for more information consult the
     [Cost Overview guide](docs/COST_OVERVIEW.md)).

> 4. **Easy to use and operate:** The solution is designed to interact directly to

Reviewer 2: The term "easy" is a bit subjective here. This section is a bit of a blend of operational excellence (robust error handling, logging etc.) and extensible by via an API

Author: Not sure here, Excellent to operate seems arrogant 😂  - what about "Simple" 😂

Reviewer 2: https://aws.amazon.com/architecture/well-architected/ - see number 2

The problem I have with terms like easy and simple is that they are subjective based on the experience of the operator

Author: Ok, I changed to `Ready to use and operate`? I think that is quite equivalent to what I'm trying to express with the paragraph?

   Amazon S3 for high compatibility with existing toolsets. It provides a REST

>    API to allow you to integrate it in your own applications, and a easy to use

Reviewer 2: let's avoid the word easy

   UI interface. When performing deletion jobs, information is provided in
   real-time to allow good observability. After the job completes, detailed
   reports are available documenting all the actions performed to individual S3
   Objects, and detailed error traces in case of failures to guide the user to

>    the troubleshooting process and easily identify remediation actions (for more

Reviewer 2: ```suggestion
   the troubleshooting process and identify remediation actions. For more
```

>    information consult the [Troubleshooting guide](docs/TROUBLESHOOTING.md)).

Reviewer 2: ```suggestion
   information consult the [Troubleshooting guide](docs/TROUBLESHOOTING.md).
```
